tests/utils/tiny_model_summary.json

```diff
@@ -0,0 +1,3181 @@
+{
```

Comment:
I think we should just have a simple way to handle that from the library's perspective, maybe from the AUTO classes -> a helper that would return you the processors & modeling class according to the model short name.

Having this here works for me to merge the PR, but I'm overall not keen on relying on such JSON files (on the Hub or not). I think this file should ideally be obtained transparently with a good API.

Reply:
The origin of this file: 
- we (me) created tiny models (v2) and upload them to Hub
  - However, we don't always get the necessary objects
    - maybe only fast tokenizer, no slow tokenizer (or maybe vice versa too?)
    - the tokenizer / feature extractor / image processor may not be created: one possible reason is the script could not find any checkpoint in order to create them
  - that json file contains information of **the result** of **tiny model creation**, i.e. which objects are created successfully.
    - **it is not** the `classess` for a specific model type that we want to create the instances (and then save/upload).
  - Therefore, `this file should ideally be obtained transparently with a good API` would mean
     - to re-run the tiny model creation script (without uploading part)
       - this is actually what we want to avoid during testing (as it is quite convoluted
       - re-run the tiny model scripts against different commits may get different set of objects
       - since we eventually want to download the uploaded files from Hub, we should use the information that is obtained when those models being created, not the information at test time
   - Regarding tiny model creation for newly add models (in a regular basis), it also requires such information (so we know what have been created or not

----------

In fact, we can avoid this fie by
- getting the `repo_id` using f`tiny-random-{model_class}`, without knowing what it contains
- using `try ... except ...` for all classes (for a model type)
  ```python
  try:
    model = model_class.from_pretrained(repo_id)
  except:
    model = None
  try:
    fast_tokenizer = fast_tokenizer_class.from_pretrained(repo_id)
  except:
    fast_tokenizer = None
  try:
    slow_tokenizer = slow_tokenizer_class.from_pretrained(repo_id)
  except:
    slow_tokenizer= None
  try:
    image_processor = image_processor _class.from_pretrained(repo_id)
  except:
    image_processor= None
  ....
  ```
    but I found having a file has more advantage over this try/except approach.